problem with compiling source code of the developmental version

I am trying to install the developmental version, but I consistently get the same error message. I re-installed R+Rstudio+Rtools, deleted the library directory, etc, but the same compilation error emerged. Here's what's happening:

```
> library(devtools)
> devtools::install_github("hadley/haven")
Downloading GitHub repo hadley/haven@master
from URL https://api.github.com/repos/hadley/haven/zipball/master
Installing haven
Installing 1 package: BH
Installing package into ‘C:/Users/endre/OneDrive/Documents/R/win-library/3.3’
(as ‘lib’ is unspecified)
trying URL 'https://cran.rstudio.com/bin/windows/contrib/3.3/BH_1.60.0-2.zip'
Content type 'application/zip' length 15529281 bytes (14.8 MB)
downloaded 14.8 MB

package ‘BH’ successfully unpacked and MD5 sums checked

The downloaded binary packages are in
	C:\Users\endre\AppData\Local\Temp\RtmpIBRdna\downloaded_packages
Installing 1 package: hms
Installing package into ‘C:/Users/endre/OneDrive/Documents/R/win-library/3.3’
(as ‘lib’ is unspecified)
trying URL 'https://cran.rstudio.com/bin/windows/contrib/3.3/hms_0.2.zip'
Content type 'application/zip' length 20720 bytes (20 KB)
downloaded 20 KB

package ‘hms’ successfully unpacked and MD5 sums checked

The downloaded binary packages are in
	C:\Users\endre\AppData\Local\Temp\RtmpIBRdna\downloaded_packages
Installing 1 package: Rcpp
Installing package into ‘C:/Users/endre/OneDrive/Documents/R/win-library/3.3’
(as ‘lib’ is unspecified)
trying URL 'https://cran.rstudio.com/bin/windows/contrib/3.3/Rcpp_0.12.6.zip'
Content type 'application/zip' length 3256650 bytes (3.1 MB)
downloaded 3.1 MB

package ‘Rcpp’ successfully unpacked and MD5 sums checked

The downloaded binary packages are in
	C:\Users\endre\AppData\Local\Temp\RtmpIBRdna\downloaded_packages
Installing 1 package: readr
Installing package into ‘C:/Users/endre/OneDrive/Documents/R/win-library/3.3’
(as ‘lib’ is unspecified)
trying URL 'https://cran.rstudio.com/bin/windows/contrib/3.3/readr_1.0.0.zip'
Content type 'application/zip' length 1164567 bytes (1.1 MB)
downloaded 1.1 MB

package ‘readr’ successfully unpacked and MD5 sums checked

The downloaded binary packages are in
	C:\Users\endre\AppData\Local\Temp\RtmpIBRdna\downloaded_packages
Installing 1 package: tibble
Installing package into ‘C:/Users/endre/OneDrive/Documents/R/win-library/3.3’
(as ‘lib’ is unspecified)
trying URL 'https://cran.rstudio.com/bin/windows/contrib/3.3/tibble_1.1.zip'
Content type 'application/zip' length 555197 bytes (542 KB)
downloaded 542 KB

package ‘tibble’ successfully unpacked and MD5 sums checked

The downloaded binary packages are in
	C:\Users\endre\AppData\Local\Temp\RtmpIBRdna\downloaded_packages
"C:/PROGRA~1/R/R-33~1.1/bin/x64/R" --no-site-file --no-environ --no-save  \
  --no-restore --quiet CMD INSTALL  \
  "C:/Users/endre/AppData/Local/Temp/RtmpIBRdna/devtools18504bee5c84/hadley-haven-94f20ed"  \
  --library="C:/Users/endre/OneDrive/Documents/R/win-library/3.3"  \
  --install-tests 

* installing *source* package 'haven' ...
** libs
c:/Rtools/mingw_64/bin/gcc  -I"C:/PROGRA~1/R/R-33~1.1/include" -DNDEBUG    -I"C:/Users/endre/OneDrive/Documents/R/win-library/3.3/Rcpp/include" -I"C:/Users/endre/OneDrive/Documents/R/win-library/3.3/BH/include" -I"d:/Compiler/gcc-4.9.3/local330/include"     -O2 -Wall  -std=gnu99 -mtune=core2 -c CKHashTable.c -o CKHashTable.o
c:/Rtools/mingw_64/bin/g++  -I"C:/PROGRA~1/R/R-33~1.1/include" -DNDEBUG    -I"C:/Users/endre/OneDrive/Documents/R/win-library/3.3/Rcpp/include" -I"C:/Users/endre/OneDrive/Documents/R/win-library/3.3/BH/include" -I"d:/Compiler/gcc-4.9.3/local330/include"     -O2 -Wall  -mtune=core2 -c DfReader.cpp -o DfReader.o
In file included from DfReader.cpp:8:0:
haven_types.h: In function 'std::string formatAttribute(FileType)':
haven_types.h:42:1: warning: control reaches end of non-void function [-Wreturn-type]
 }
 ^
DfReader.cpp: In member function 'readstat_off_t DfReaderInputStream<Stream>::seek(readstat_off_t, readstat_io_flags_t, void*) [with Stream = std::basic_istringstream<char>; readstat_off_t = long long int; readstat_io_flags_t = readstat_io_flags_e]':
DfReader.cpp:436:5: warning: 'dir' may be used uninitialized in this function [-Wmaybe-uninitialized]
     file_.seekg(offset, dir);
     ^
DfReader.cpp: In member function 'readstat_off_t DfReaderInputStream<Stream>::seek(readstat_off_t, readstat_io_flags_t, void*) [with Stream = std::basic_ifstream<char>; readstat_off_t = long long int; readstat_io_flags_t = readstat_io_flags_e]':
DfReader.cpp:436:5: warning: 'dir' may be used uninitialized in this function [-Wmaybe-uninitialized]
c:/Rtools/mingw_64/bin/g++  -I"C:/PROGRA~1/R/R-33~1.1/include" -DNDEBUG    -I"C:/Users/endre/OneDrive/Documents/R/win-library/3.3/Rcpp/include" -I"C:/Users/endre/OneDrive/Documents/R/win-library/3.3/BH/include" -I"d:/Compiler/gcc-4.9.3/local330/include"     -O2 -Wall  -mtune=core2 -c DfWriter.cpp -o DfWriter.o
DfWriter.cpp: In member function 'const char* Writer::var_format(Rcpp::RObject, VarType)':
DfWriter.cpp:179:3: warning: control reaches end of non-void function [-Wreturn-type]
   }
   ^
In file included from DfWriter.cpp:4:0:
haven_types.h: In function 'std::string formatAttribute(FileType)':
haven_types.h:42:1: warning: control reaches end of non-void function [-Wreturn-type]
 }
 ^
c:/Rtools/mingw_64/bin/g++  -I"C:/PROGRA~1/R/R-33~1.1/include" -DNDEBUG    -I"C:/Users/endre/OneDrive/Documents/R/win-library/3.3/Rcpp/include" -I"C:/Users/endre/OneDrive/Documents/R/win-library/3.3/BH/include" -I"d:/Compiler/gcc-4.9.3/local330/include"     -O2 -Wall  -mtune=core2 -c RcppExports.cpp -o RcppExports.o
c:/Rtools/mingw_64/bin/gcc  -I"C:/PROGRA~1/R/R-33~1.1/include" -DNDEBUG    -I"C:/Users/endre/OneDrive/Documents/R/win-library/3.3/Rcpp/include" -I"C:/Users/endre/OneDrive/Documents/R/win-library/3.3/BH/include" -I"d:/Compiler/gcc-4.9.3/local330/include"     -O2 -Wall  -std=gnu99 -mtune=core2 -c readstat_bits.c -o readstat_bits.o
c:/Rtools/mingw_64/bin/gcc  -I"C:/PROGRA~1/R/R-33~1.1/include" -DNDEBUG    -I"C:/Users/endre/OneDrive/Documents/R/win-library/3.3/Rcpp/include" -I"C:/Users/endre/OneDrive/Documents/R/win-library/3.3/BH/include" -I"d:/Compiler/gcc-4.9.3/local330/include"     -O2 -Wall  -std=gnu99 -mtune=core2 -c readstat_convert.c -o readstat_convert.o
readstat_convert.c: In function 'readstat_convert':
readstat_convert.c:24:42: warning: passing argument 2 of 'libiconv' from incompatible pointer type
         size_t status = iconv(converter, (readstat_iconv_inbuf_t)&src, &src_len, &dst_end, &dst_left);
                                          ^
In file included from readstat_iconv.h:1:0,
                 from readstat_convert.c:4:
C:/PROGRA~1/R/R-33~1.1/include/iconv.h:51:15: note: expected 'const char **' but argument is of type 'char **'
 #define iconv libiconv
               ^
C:/PROGRA~1/R/R-33~1.1/include/iconv.h:53:8: note: in expansion of macro 'iconv'
 size_t iconv (iconv_t cd, const char* * inbuf, size_t *inbytesleft, 
        ^
c:/Rtools/mingw_64/bin/gcc  -I"C:/PROGRA~1/R/R-33~1.1/include" -DNDEBUG    -I"C:/Users/endre/OneDrive/Documents/R/win-library/3.3/Rcpp/include" -I"C:/Users/endre/OneDrive/Documents/R/win-library/3.3/BH/include" -I"d:/Compiler/gcc-4.9.3/local330/include"     -O2 -Wall  -std=gnu99 -mtune=core2 -c readstat_error.c -o readstat_error.o
c:/Rtools/mingw_64/bin/gcc  -I"C:/PROGRA~1/R/R-33~1.1/include" -DNDEBUG    -I"C:/Users/endre/OneDrive/Documents/R/win-library/3.3/Rcpp/include" -I"C:/Users/endre/OneDrive/Documents/R/win-library/3.3/BH/include" -I"d:/Compiler/gcc-4.9.3/local330/include"     -O2 -Wall  -std=gnu99 -mtune=core2 -c readstat_io_unistd.c -o readstat_io_unistd.o
c:/Rtools/mingw_64/bin/gcc  -I"C:/PROGRA~1/R/R-33~1.1/include" -DNDEBUG    -I"C:/Users/endre/OneDrive/Documents/R/win-library/3.3/Rcpp/include" -I"C:/Users/endre/OneDrive/Documents/R/win-library/3.3/BH/include" -I"d:/Compiler/gcc-4.9.3/local330/include"     -O2 -Wall  -std=gnu99 -mtune=core2 -c readstat_parser.c -o readstat_parser.o
c:/Rtools/mingw_64/bin/gcc  -I"C:/PROGRA~1/R/R-33~1.1/include" -DNDEBUG    -I"C:/Users/endre/OneDrive/Documents/R/win-library/3.3/Rcpp/include" -I"C:/Users/endre/OneDrive/Documents/R/win-library/3.3/BH/include" -I"d:/Compiler/gcc-4.9.3/local330/include"     -O2 -Wall  -std=gnu99 -mtune=core2 -c readstat_value.c -o readstat_value.o
c:/Rtools/mingw_64/bin/gcc  -I"C:/PROGRA~1/R/R-33~1.1/include" -DNDEBUG    -I"C:/Users/endre/OneDrive/Documents/R/win-library/3.3/Rcpp/include" -I"C:/Users/endre/OneDrive/Documents/R/win-library/3.3/BH/include" -I"d:/Compiler/gcc-4.9.3/local330/include"     -O2 -Wall  -std=gnu99 -mtune=core2 -c readstat_variable.c -o readstat_variable.o
c:/Rtools/mingw_64/bin/gcc  -I"C:/PROGRA~1/R/R-33~1.1/include" -DNDEBUG    -I"C:/Users/endre/OneDrive/Documents/R/win-library/3.3/Rcpp/include" -I"C:/Users/endre/OneDrive/Documents/R/win-library/3.3/BH/include" -I"d:/Compiler/gcc-4.9.3/local330/include"     -O2 -Wall  -std=gnu99 -mtune=core2 -c readstat_writer.c -o readstat_writer.o
c:/Rtools/mingw_64/bin/gcc  -I"C:/PROGRA~1/R/R-33~1.1/include" -DNDEBUG    -I"C:/Users/endre/OneDrive/Documents/R/win-library/3.3/Rcpp/include" -I"C:/Users/endre/OneDrive/Documents/R/win-library/3.3/BH/include" -I"d:/Compiler/gcc-4.9.3/local330/include"     -O2 -Wall  -std=gnu99 -mtune=core2 -c tagged_na.c -o tagged_na.o
c:/Rtools/mingw_64/bin/g++ -shared -s -static-libgcc -o haven.dll tmp.def CKHashTable.o DfReader.o DfWriter.o RcppExports.o readstat_bits.o readstat_convert.o readstat_error.o readstat_io_unistd.o readstat_parser.o readstat_value.o readstat_variable.o readstat_writer.o tagged_na.o -lRiconv -Ld:/Compiler/gcc-4.9.3/local330/lib/x64 -Ld:/Compiler/gcc-4.9.3/local330/lib -LC:/PROGRA~1/R/R-33~1.1/bin/x64 -lR
DfReader.o:DfReader.cpp:(.text$_Z12df_parse_sasI17DfReaderInputFileEN4Rcpp6VectorILi19ENS1_15PreserveStorageEEES4_S4_Ss[_Z12df_parse_sasI17DfReaderInputFileEN4Rcpp6VectorILi19ENS1_15PreserveStorageEEES4_S4_Ss]+0x2cd): undefined reference to `readstat_parse_sas7bcat'
DfReader.o:DfReader.cpp:(.text$_Z12df_parse_sasI17DfReaderInputFileEN4Rcpp6VectorILi19ENS1_15PreserveStorageEEES4_S4_Ss[_Z12df_parse_sasI17DfReaderInputFileEN4Rcpp6VectorILi19ENS1_15PreserveStorageEEES4_S4_Ss]+0x3d4): undefined reference to `readstat_parse_sas7bdat'
DfReader.o:DfReader.cpp:(.text$_Z12df_parse_sasI16DfReaderInputRawEN4Rcpp6VectorILi19ENS1_15PreserveStorageEEES4_S4_Ss[_Z12df_parse_sasI16DfReaderInputRawEN4Rcpp6VectorILi19ENS1_15PreserveStorageEEES4_S4_Ss]+0x2ce): undefined reference to `readstat_parse_sas7bcat'
DfReader.o:DfReader.cpp:(.text$_Z12df_parse_sasI16DfReaderInputRawEN4Rcpp6VectorILi19ENS1_15PreserveStorageEEES4_S4_Ss[_Z12df_parse_sasI16DfReaderInputRawEN4Rcpp6VectorILi19ENS1_15PreserveStorageEEES4_S4_Ss]+0x3ad): undefined reference to `readstat_parse_sas7bdat'
DfReader.o:DfReader.cpp:(.text$_Z12df_parse_dtaI17DfReaderInputFileEN4Rcpp6VectorILi19ENS1_15PreserveStorageEEES4_Ss[_Z12df_parse_dtaI17DfReaderInputFileEN4Rcpp6VectorILi19ENS1_15PreserveStorageEEES4_Ss]+0x47b): undefined reference to `readstat_parse_dta'
DfReader.o:DfReader.cpp:(.text$_Z12df_parse_dtaI16DfReaderInputRawEN4Rcpp6VectorILi19ENS1_15PreserveStorageEEES4_Ss[_Z12df_parse_dtaI16DfReaderInputRawEN4Rcpp6VectorILi19ENS1_15PreserveStorageEEES4_Ss]+0x25a): undefined reference to `readstat_parse_dta'
DfReader.o:DfReader.cpp:(.text$_Z13df_parse_spssI17DfReaderInputFileEN4Rcpp6VectorILi19ENS1_15PreserveStorageEEES4_bb[_Z13df_parse_spssI17DfReaderInputFileEN4Rcpp6VectorILi19ENS1_15PreserveStorageEEES4_bb]+0x26d): undefined reference to `readstat_parse_por'
DfReader.o:DfReader.cpp:(.text$_Z13df_parse_spssI17DfReaderInputFileEN4Rcpp6VectorILi19ENS1_15PreserveStorageEEES4_bb[_Z13df_parse_spssI17DfReaderInputFileEN4Rcpp6VectorILi19ENS1_15PreserveStorageEEES4_bb]+0x391): undefined reference to `readstat_parse_sav'
DfReader.o:DfReader.cpp:(.text$_Z13df_parse_spssI16DfReaderInputRawEN4Rcpp6VectorILi19ENS1_15PreserveStorageEEES4_bb[_Z13df_parse_spssI16DfReaderInputRawEN4Rcpp6VectorILi19ENS1_15PreserveStorageEEES4_bb]+0x26d): undefined reference to `readstat_parse_por'
DfReader.o:DfReader.cpp:(.text$_Z13df_parse_spssI16DfReaderInputRawEN4Rcpp6VectorILi19ENS1_15PreserveStorageEEES4_bb[_Z13df_parse_spssI16DfReaderInputRawEN4Rcpp6VectorILi19ENS1_15PreserveStorageEEES4_bb]+0x366): undefined reference to `readstat_parse_sav'
DfWriter.o:DfWriter.cpp:(.text$_ZN6Writer5writeEv[_ZN6Writer5writeEv]+0xfc6): undefined reference to `readstat_begin_writing_sav'
DfWriter.o:DfWriter.cpp:(.text$_ZN6Writer5writeEv[_ZN6Writer5writeEv]+0xfee): undefined reference to `readstat_begin_writing_sas7bdat'
DfWriter.o:DfWriter.cpp:(.text$_ZN6Writer5writeEv[_ZN6Writer5writeEv]+0x100f): undefined reference to `readstat_begin_writing_dta'
collect2.exe: error: ld returned 1 exit status
no DLL was created
ERROR: compilation failed for package 'haven'
* removing 'C:/Users/endre/OneDrive/Documents/R/win-library/3.3/haven'
Error: Command failed (1)
```

--- a/src/Makevars
+++ b/src/Makevars
@@ -2,4 +2,6 @@ CFILES = $(wildcard *.c sas/*.c spss/*.c stata/*.c)
 CPPFILES = $(wildcard *.cpp)
 
 SOURCES = $(CFILES) $(CPPFILES)
+
+# This must be defined identically in Makevars.win
 OBJECTS = $(CFILES:.c=.o) $(CPPFILES:.cpp=.o)

--- a/src/Makevars.win
+++ b/src/Makevars.win
@@ -1,1 +1,6 @@
+include Makevars
+
+# This is also defined in Makevars, but somehow the definition from there is not used
+OBJECTS = $(CFILES:.c=.o) $(CPPFILES:.cpp=.o)
+
 PKG_LIBS=-lRiconv

